tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -298,20 +310,24 @@ def __init__(self, test_ctx, *args, **kwargs):
                 'kafka_connection_rate_limit': 10000,
                 'kafka_connections_max': 50000,
 
-                # Enable segment size jitter as this is a stress test and does not
-                # rely on exact segment counts.
-                'log_segment_size_jitter_percent': 5,
-
                 # In testing tiered storage, we care about creating as many
                 # cloud segments as possible. To that end, bounding the segment
                 # size isn't productive.
                 'cloud_storage_segment_size_min': 1,
                 'log_segment_size_min': 1024,
 
+                # Use more connections in testing to reach reasonable throughput
+                # number even for small segments and high S3 latency.
+                'cloud_storage_max_connections': 100,
```

Comment:
Ah so the comment above is already out of date?

Reply:
I removed this. It didn't help. I'm making the performance investigation out of scope of this PR. Once we have the test running we can work on improving the performance.